Sources/MMIO/Documentation.docc/Getting-Started/Accessing-Registers.md

```diff
@@ -0,0 +1,174 @@
+
+The `modify` method is the preferred way to update register values, especially when you only want to change specific bit fields while leaving others untouched. It performs a read-modify-write sequence that is atomic with respect to the current software thread's execution (i.e., no other code in the same thread can interleave). However, it is **not** inherently atomic with respect to hardware interrupts modifying the same register. For true hardware atomicity, interrupt masking (e.g., disabling and re-enabling interrupts) may be required around the `modify` call.
+
+The sequence is:
+1. Read the current value from hardware.
+2. Provide this value (as a `Read` view) and an `inout Write` view (initialized from the read value) to your closure.
+3. Inside the closure, you modify the `Write` view's fields.
+4. After the closure, the modified `Write` view is written back to hardware.
+
+This preserves unrelated bits, crucial for many register interactions.
+
+```swift
+timer0.control.modify { currentStatus, newStatus in
+  // currentStatus: 'Read' view of the register's state before modification.
+  // newStatus: 'inout Write' view, initialized with currentStatus, for modifications.
+
+  // Assuming 'enable' and 'interruptEnable' are projected as Bool
+  newStatus.enable = true
+  newStatus.interruptEnable = false
+
+  // Assuming 'prescaler' is a raw segment, set it via .raw
+  newStatus.raw.prescaler = 15
+
+  // For a WriteOnly field like 'clearInterrupt':
+  // If projected (e.g., @WriteOnly(bits: ..., as: Bool.self)):
+  // newStatus.clearInterrupt = true
+  // If not projected, or for direct raw access:
+  newStatus.raw.clearInterrupt = 1
+  // Fields not explicitly set in newStatus retain their values from
+  // currentStatus.
+}
+```
+
+If a register's `Read` and `Write` views are identical (all fields are ``MMIO/ReadWrite(bits:as:)`` or ``MMIO/Reserved(bits:as:)``), a simpler `modify` signature is available:
+
+```swift
+// Assuming 'symmetricReg.control' only has ReadWrite/Reserved fields:
+symmetricReg.control.modify { view in
+  // 'view' is an inout ReadWrite view, initialized with the current register value.
+  view.someProjectedField = newValue
+  view.raw.someRawField = anotherValue
+}
+```
+
+The two-parameter `modify` closure is essential for registers with asymmetric fields (those defined with ``MMIO/ReadOnly(bits:as:)`` or ``MMIO/WriteOnly(bits:as:)``), as it allows you to inspect read-only state while preparing writes.
+
+### 2. Using `write` (For Overwriting the Entire Register)
+
+The `write` method directly overwrites the entire register. This is appropriate when you need to set all fields to known states simultaneously or/and need to avoid a read of the hardware.
+
+You can construct the `Write` view for `write` in two main ways:
+
+```swift
+// Option 1: Initialize a zeroed Write view and set fields (often clearer)
+timer0.control.write { controlToWrite in
+    // controlToWrite starts with all bits as 0.
```

Comment:
This is a very easy to miss property (that all bits are 0 to start with), and it's only present in documentation, and not implied by the API. I wonder if we have any options of highlighting it further -- i.e. maybe somehow making the API name carry that information, or requiring that the user manually sets the default value, or something like that? (Context: https://github.com/apple/swift-mmio/issues/152)